### PR TITLE
add io.github.lemonade_emu.Lemonade

### PR DESCRIPTION
**Pull request by Ishan09811**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by Ishan09811** [comment]
bot, build org.lemonade_emu.lemonade

**Comment by Ishan09811** [comment]
bot, build org.lemonade_emu.lemonade

**Comment by hfiguiere** [comment]
>  Please describe your application briefly.

this is missing :up:

File: org.lemonade_emu.lemonade.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        "/lib/cmake",
+        "/share/doc",
+        "/share/man",
+        "/src",
+        "*.a",
+        "*.la"
+    ],
+    "modules": [
+        "shared-modules/libusb/libusb.json",
+        "shared-modules/SDL2/SDL2-with-libdecor.json",
```
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.json:49]: sdl with libdecor is in the runtime since 23.08

File: org.lemonade_emu.lemonade.json [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            ],
+            "post-install": [
+                "install -Dm755 ../lemonade-launcher.sh /app/bin/lemonade-launcher",
+                "install -Dm644 ../org.lemonade_emu.lemonade.metainfo.xml /app/share/appdata/org.lemonade_emu.lemonade.metainfo.xml",
+                "desktop-file-install --dir=/app/share/applications ../dist/lemonade-qt.desktop",
+                "desktop-file-edit --set-key StartupWMClass --set-value lemonade-qt --set-key Exec --set-value lemonade-launcher /app/share/applications/lemonade-qt.desktop",
+                "install -Dm644 ../dist/lemonade.svg /app/share/icons/hicolor/scalable/apps/lemonade.svg",
+                "install -Dm644 ../dist/icon.png /app/share/icons/hicolor/512x512/apps/lemonade.png",
+                "mv /app/share/mime/packages/lemonade.xml /app/share/mime/packages/org.lemonade_emu.lemonade.xml",
+                "sed 's/lemonade/org.lemonade_emu.lemonade/g' -i /app/share/mime/packages/org.lemonade_emu.lemonade.xml"
```
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.json:146]: Please see the thread https://github.com/flathub/flathub/pull/5133#discussion_r1552199625

The is of sed is always strongly discourage.

**Comment by Ishan09811** [comment]
@hfiguiere 
I already described the application in ``org.lemonade_emu.lemonade.metainfo.xml``

File: org.lemonade_emu.lemonade.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.lemonade_emu.lemonade</id>
+  <name>Lemonade</name>
+  <summary>An Open Source Nintendo 3DS emulator</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0</project_license>
+  <content_rating type="oars-1.0">
```
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.metainfo.xml:8]:
```suggestion
  <content_rating type="oars-1.1">
```

File: org.lemonade_emu.lemonade.json [hunk 4]
```diff
@@ -0,0 +1,172 @@
+{
+    "app-id": "org.lemonade_emu.lemonade",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "6.6",
```
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.json:4]: runtime 6.7 is just out.

File: screenshots/lemonade_flathub-1.png
Comment (hfiguiere) [inline screenshots/lemonade_flathub-1.png:1]: this screenshpt doesn't look very Linux...

**Comment by hfiguiere** [comment]
> @hfiguiere I already described the application in `org.lemonade_emu.lemonade.metainfo.xml`

You like making things harder for reviewer? Or is your package so so special?

File: screenshots/lemonade_flathub-1.png
Comment (kleidiss) [reply screenshots/lemonade_flathub-1.png:1]: It's a custom QStyleSheet , Is there a problem with that , i can revert it

Comment (hfiguiere) [reply screenshots/lemonade_flathub-1.png:1]: "c:\" is NOT the stylesheet. It's Windows.

**Comment by Ishan09811** [comment]
Lol,

File: screenshots/lemonade_flathub-1.png
Comment (kleidiss) [reply screenshots/lemonade_flathub-1.png:1]: Oh i thought you meant the design , yeah i took the screenshots on windows , i am away from my linux machine for some days , i will update a soon as i can

**Comment by hfiguiere** [comment]
> @hfiguiere Lol, ok let me make a readme

All that is asked is where there is "please decsribe" and you acknowledge that there is actually a short description of the package. No need to write a novel.

File: org.lemonade_emu.lemonade.metainfo.xml
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.metainfo.xml:1]: this file should be managed upstream.

File: org.lemonade_emu.lemonade.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  <provides>
+    <binary>lemonade-qt</binary>
+    <id>lemonade-qt.desktop</id>
+  </provides>
+  <url type="homepage">https://lemonade-emu.github.io/</url>
+  <url type="contact">https://discord.gg/kKKantKn</url>
+  <screenshots>
+    <screenshot type="default"><image>https://github.com/Lemonade-emu/Lemonade/blob/master/assets/lemonade_flathub-2.png</image></screenshot>
+    <screenshot><image>https://github.com/Lemonade-emu/Lemonade/blob/master/assets/lemonade_flathub-1.png</image></screenshot>
+    <screenshot><image>https://github.com/Lemonade-emu/Lemonade/blob/master/assets/lemonade_flathub-3.png</image></screenshot>
```
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.metainfo.xml:25]: 1. this link don't link to the image.
2. they should NEVER point at a branch but at a specific commit

File: screenshots/lemonade_flathub-1.png
Comment (hfiguiere) [reply screenshots/lemonade_flathub-1.png:1]: also why are the screenshots in this repository?

File: org.lemonade_emu.lemonade.metainfo.xml (hunk 5 above)
Comment (kleidiss) [reply org.lemonade_emu.lemonade.metainfo.xml:25]: Im fixing that , will push shortly

File: org.lemonade_emu.lemonade.json [hunk 6]
```diff
@@ -0,0 +1,172 @@
+{
+    "app-id": "org.lemonade_emu.lemonade",
```
Comment (hfiguiere) [inline org.lemonade_emu.lemonade.json:2]: does upstream really control `lemonade-emu.org`. I see nothing there.

File: screenshots/lemonade_flathub-1.png
Comment (bbhtt) [inline screenshots/lemonade_flathub-1.png:1]: Screenshots should be upstream

File: .github/workflows/merge.js
Comment (bbhtt) [inline .github/workflows/merge.js:1]: This seems to be from https://github.com/flathub/org.yuzu_emu.yuzu/blob/master/.github/workflows/merge.js, in that case you should know that yuzu spammed Flathub buildserver with duplicate builds due to this script pushing extra commits in a PR.

Please drop this and the auto merge workflow.

Comment (kleidiss) [reply .github/workflows/merge.js:1]: They were from Citra's flathub repo, deleted

**Comment by Ishan09811** [comment]
@hfiguiere 
why we got blocked label?

**Comment by bbhtt** [comment]
> @hfiguiere why we got blocked label?

Because the domain you are using in the app id does not seem legitimate and controlled by the project.

**Comment by kleidiss** [comment]
> @hfiguiere why we got blocked label?

Because of the workflows , i removed them so we should be good

File: assets/lemonade_flathub-1.png
Comment (hfiguiere) [inline assets/lemonade_flathub-1.png:1]: why are screenshot TWICE in the repo.

I'll reiterate: they should be managed upstream.

**Comment by hfiguiere** [comment]
> @hfiguiere why we got blocked label?

because we try to maximize our efficiency by marking thanigs that can't be merged yet. Note that if there are still unanswered questions that's probably one of the reasons.

Also note you ask the wrong person:
image

(but I'm agreement)

**Comment by bbhtt** [comment]
We need to decide internally on what to do with Yuzu/Citra forks, given the recent legal situation. Please wait until then, and someone will update the status labels accordinginly.

**Comment by kleidiss** [comment]
> We need to decide internally on what to do with Yuzu/Citra forks, given the recent legal situation. Please wait until then, and someone will update the status labels accordinginly.

Yeah i understand, After all hosting a fork of a project that got taken down will always be a question, I will fix the pointed issues in the review and leave it up to you guys to decide, take your time.

File: assets/lemonade_flathub-1.png
Comment (kleidiss) [reply assets/lemonade_flathub-1.png:1]: Can you please elaborate on what you mean by upstream in this case? 

Sorry for being dumb

File: org.lemonade_emu.lemonade.json (hunk 6 above)
Comment (Ishan09811) [reply org.lemonade_emu.lemonade.json]: @hfiguiere 
now we changed it to io.github.lemonade-emu.Lemonade as our website https://lemonade-emu.github.io/

**Comment by kleidiss** [comment]
> I'm not particularly concerned about the legal implications of this, but I can't help but notice there's also another fork called Lime3DS (submitted to Flathub as well). What's the difference between yours and the other one?

We are 2 different teams which have put our own changes into the Citra code 

You can look at the releases for both projects and see the changes made to the base Citra Emulator 

Now if you only can choose one , idk what to tell you

**Comment by Ishan09811** [comment]
bot, build io.github.lemonade_emu.Lemonade

**Comment by Ishan09811** [comment]
bot, build io.github.lemonade_emu.Lemonade

**Comment by kleidiss** [comment]
bot, build io.github.lemonade_emu.Lemonade

**Comment by kleidiss** [comment]
bot, build io.github.lemonade_emu.Lemonade

**Comment by kleidiss** [comment]
bot, build io.github.lemonade_emu.Lemonade

**Comment by kleidiss** [comment]
Is anything else left for me to do?

**Comment by Ishan09811** [comment]
@hfiguiere 
Is anything else left for us to do?

File: io.github.lemonade_emu.Lemonade.json [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+        "lemonade"
+    ],
+    "copy-icon": true,
+    "build-options": {
+        "env": {
+            "CI": "1",
+            "GITHUB_ACTIONS": "1",
+            "GITHUB_REPOSITORY": "Lemonade-emu/Lemonade",
+            "GITHUB_REF_NAME": "master"
+        },
```
Comment (bbhtt) [inline io.github.lemonade_emu.Lemonade.json:23]: What is this part for?

File: screenshots/flathub-1.png
Comment (bbhtt) [inline screenshots/flathub-1.png:1]: These screenshots doesn't seem used? https://raw.githubusercontent.com/Lemonade-emu/lemonade-flatpak/362eba9e75274a10bed6d8e3eff873aca85603c8/screenshots/flathub-3.png is coming from upstream

File: io.github.lemonade_emu.Lemonade.metainfo.xml
Comment (bbhtt) [inline io.github.lemonade_emu.Lemonade.metainfo.xml:1]: metainfo file should be upstream

Comment (Ishan09811) [reply io.github.lemonade_emu.Lemonade.metainfo.xml:1]: @bbhtt 
any reason?

Comment (bbhtt) [reply io.github.lemonade_emu.Lemonade.metainfo.xml:1]: It's not specific to flatpak, you'll need it for any other format. Also allows translation and contributing when upstream

Comment (Ishan09811) [reply io.github.lemonade_emu.Lemonade.metainfo.xml:1]: @bbhtt 
sorry for being dumb but first of all what do you mean upstream?

Comment (bbhtt) [reply io.github.lemonade_emu.Lemonade.metainfo.xml:1]: Somewhere under here https://github.com/Lemonade-emu/

File: screenshots/flathub-1.png
Comment (Ishan09811) [reply screenshots/flathub-1.png:1]: @bbhtt 
yes, we provided the screenshots from our main repo

File: io.github.lemonade_emu.Lemonade.json (hunk 7 above)
Comment (Ishan09811) [reply io.github.lemonade_emu.Lemonade.json:23]: @bbhtt 
This section specifies environment variables and build options to be used during the build process.

File: screenshots/flathub-1.png
Comment (bbhtt) [reply screenshots/flathub-1.png:1]: remove them from here then

File: io.github.lemonade_emu.Lemonade.json (hunk 7 above)
Comment (bbhtt) [reply io.github.lemonade_emu.Lemonade.json:23]: I know but why do you need Github CI environment variables here for? The build pipeline does not run on Github CI here and having them in manifest won't affect anything.


Are you using them in the buildsystem then?

Comment (Ishan09811) [reply io.github.lemonade_emu.Lemonade.json:23]: @bbhtt 
no, this is already from Citra json so we didn't change it btw if it's not required or won't affect anything we can remove this section.

**Comment by Ishan09811** [comment]
@bbhtt 
Is anything else left for us to do?

**Comment by bbhtt** [comment]
bot, build io.github.lemonade_emu.Lemonade

**Comment by OpenSauce04** [comment]
I would advise against adding this app for the moment, as the Lemonade project is currently experiencing a leadership crisis and its future is uncertain

**Comment by Ishan09811** [comment]
@OpenSauce04 
Me and Gamer64ytb decided to continue the project

**Comment by SomeUserOnGithub63** [comment]
Yeah sure the Lemonade repository got deleted

**Comment by bbhtt** [comment]
Indeed thanks.

**Comment by Ishan09811** [comment]
Yes, it's got deleted by us and we will create a new repo

**Pull request closed without merging**